src/poetry/console/commands/plugin/plugin_command_mixin.py

```diff
@@ -0,0 +1,125 @@
+import os
+
+from typing import TYPE_CHECKING
+from typing import List
+
+import tomlkit
+
+from poetry.packages.locker import BaseLocker
+from poetry.utils.helpers import canonicalize_name
+
+
+if TYPE_CHECKING:
+    from cleo.io.io import IO
+    from entrypoints import EntryPoint
+
+    from poetry.core.packages.project_package import ProjectPackage
+    from poetry.utils.env import Env
+
+
+class PluginCommandMixin:
+    def create_env_package(
+        self, env: "Env", existing_plugins: dict
+    ) -> "ProjectPackage":
+        from poetry.core.packages.project_package import ProjectPackage
```

Comment:
Not sure if the scoped imports here are necessary, or wise. It feels like a premature optimization that adds significant mental overhead and complexity. See comments elsewhere in the review.